hazelcast/src/main/java/com/hazelcast/internal/partition/operation/PartitionReplicaSyncRequestOffloadable.java

```diff
@@ -216,20 +218,30 @@ public void start() throws Exception {
             }
 
             try {
-                // executed on generic operation thread
-                Integer permits = getPermits();
-                if (permits == null) {
-                    return;
-                }
-
-                sendOperationsForNamespaces(permits);
-                // send retry response for remaining namespaces
-                if (!namespaces.isEmpty()) {
-                    logNotEnoughPermits();
+                nodeEngine.getExecutionService().execute(ExecutionService.ASYNC_EXECUTOR,
+                        () -> {
```

Comment:
I'd prefer also moving the migration flags setting into this async executor task so that we can also offload that execution. Even if it's a short operation, currently the priority/generic operation thread is waiting for it to complete. Also, if that async executor doesn't schedule this sync execution for a long time, the other operation can be rejected because of this flags are already set.

Reply:
good point, done in https://github.com/hazelcast/hazelcast/pull/20813/commits/41afa654175ccea365f4cfce91c2c428a8a44606

Reply:
Also EE tests completed successfully with latest commit: https://jenkins.hazelcast.com/job/Hazelcast-EE-octopus-pr-builder/622/